### PR TITLE
Add @datagouv/components CI

### PR DESCRIPTION
**Pull request by nicolaskempf57**
Closes https://github.com/datagouv/data.gouv.fr/issues/1265

### CODE REVIEW

File: udata_front/theme/gouvfr/datagouv-components/README.md [hunk 1]
```diff
@@ -36,3 +36,43 @@ window.dsfr.start(); // nothing here if you're using vue-dsfr
...
+- `dev`, for our development versions based on `master`
+
+`dev` versions are done automatically and there is nothing to do.
+For our releases, a manual update of the version number is necessary. To do so, you can run :
+
+```
+npm run update-version -- NEW_VERSION
+```
+
+You should use `major`, `minor`, `patch` to let NPM compute the next version number.
```
Comment (maudetes) [inline udata_front/theme/gouvfr/datagouv-components/README.md:55]:
```suggestion
`NEW_VERSION` should be either `major`, `minor` or `patch` to let NPM compute the next version number.
```

File: udata_front/theme/gouvfr/datagouv-components/README.md [hunk 2]
```diff
@@ -36,3 +36,43 @@ window.dsfr.start(); // nothing here if you're using vue-dsfr
 ## env variables
 
 You can customize how this package work with env variables during build and runtime (with `setupComponents` function).
+
+## Publish a new version
```
Comment (maudetes) [inline udata_front/theme/gouvfr/datagouv-components/README.md:40]: Thank you for this new section :pray:

File: .circleci/config.yml [hunk 3]
```diff
@@ -196,6 +196,64 @@ jobs:
...
+          command: echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" >> .npmrc
+          working_directory: ./udata_front/theme/gouvfr/datagouv-components
+      - run:
+          name: Publish on NPM
+          working_directory: ./udata_front/theme/gouvfr/datagouv-components
+          command: |
+            if [[ $CIRCLE_TAG ]]; then
+              # This is a tagged release
+              npm run publish-stable
+              npm run update-version -- prerelease --preid=dev
```
Comment (maudetes) [inline .circleci/config.yml:248]: The update-version in CI aren't pushed back on git right?

Comment (nicolaskempf57) [reply .circleci/config.yml:248]: No the current `dev` version isn't stored anywhere outside of NPM

File: .circleci/config.yml [hunk 4]
```diff
@@ -196,6 +196,64 @@ jobs:
...
+              npm run publish-dev
+            else
+              # This is a simple development build
+              npm run update-version -- --allow-same-version $(npm view @datagouv/components@dev version)
```
[the remark this answers is not in the document]
Comment (nicolaskempf57) [reply .circleci/config.yml:252]: This line changes the package.json version to the last `dev` version published, then below we increment it by 1. As you find out [above](https://github.com/datagouv/udata-front/pull/407/files/e7cd54bc4f3b821ce655ae209eda62ab067c0ac1#r1616069150), the build number is not pushed back to git.

**Review by maudetes — APPROVED** [review]
Thank you for the explanation. All clear for me!

**Pull request merged**